incubator/hnc/test/conversion/conversion_test.go

```diff
@@ -282,6 +273,105 @@ spec:
 		MustRun("kubectl -n", nsA, "create secret generic my-creds-2 --from-literal=password=iama")
 		RunShouldContainMultiple([]string{"my-creds-1", "my-creds-2"}, propagationTime, "kubectl get secrets -n", nsB)
 	})
+
+	It("should convert inheritedFrom label to 'inherited-from' on propagated objects", func() {
+		// Before conversion, create a propagated object in v1alpha1.
+		// Create a tree a <- b, where a is the parent.
+		createSampleV1alpha1ParentChildTree()
+
+		// Create a role in namespace A that should be propagated to namespace B.
+		roleA := "team-a-sre"
+		createRole(roleA, nsA)
+		// Verify the v1a1 inheritedFrom label - `inheritedFrom`.
+		FieldShouldContain("roles", nsB, roleA, ".metadata.labels", "hnc.x-k8s.io/inheritedFrom:"+nsA)
+
+		// Convert
+		setupV1alpha2()
+
+		// Verify the converted v1a2 inheritedFrom label - `inherited-from`.
+		FieldShouldContain("roles", nsB, roleA, ".metadata.labels", "hnc.x-k8s.io/inherited-from:"+nsA)
+		// Verify the v1a1 label is gone.
+		FieldShouldNotContain("roles", nsB, roleA, ".metadata.labels", "inheritedFrom")
+	})
+	// 3 more corner cases for v1a1 propagated objects with inheritedFrom label
+	// during conversion:
+	// 1) It may not clear v1a1 propagated objects if the source object is removed right before conversion;
```

Comment:
It's not really "right before conversion" anymore, right? It's more like "after v0.5 stops but before v0.6 starts"

Or maybe "during conversion" is a better description.

Reply:
Agreed "during conversion" is better, updated.